[ADD] Account Regularizations. …

Roodin opened the pull request:
Migrated from 5.0 previous work from Acysos and Zikzakmedia.
With this addon you can preconfigure account regularizations in perodical proceesing and, executing a wizard, create the move that balance the accounts you need automatically.  A typical example is regularizations of tax accounts.

Roodin committed:
[ADD] Account Regularizations. Migrated from 5.0 previous work form Acysos an Zikzakmedia

sbidoul commented on account_regularization/__openerp__.py:
```diff
@@ -0,0 +1,55 @@
+    transfer the difference to a new account. This is used,
+    for example with tax declarations or in some countries to create
+    the 'Profit and Loss' regularization
+""",
```
Can you please move the description to README.rst file following this template: https://github.com/OCA/maintainer-tools/tree/master/template/module

sbidoul commented:
Hi, thanks for this contribution!

It seems there is duplicate code in models/account.py and models/account_regularization.py.

Roodin commented:
Sorry, you are rigth, and very fast in deed . I renamed the file but not
deleted in git :(

2015-06-26 19:10 GMT+02:00 Stéphane Bidoul (ACSONE) <
notifications@github.com>:

> Hi, thanks for this contribution!
> 
> It seems there is duplicate code in models/account.py and
> models/account_regularization.py.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/OCA/account-financial-tools/pull/247#issuecomment-115770312
> .

max3903 commented:
CLA is fixed and OK.

pedrobaeza commented on account_regularization/__init__.py:
```diff
@@ -0,0 +1,26 @@
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (c) 2008 Zikzakmedia S.L. (http://zikzakmedia.com)
+#  All Rights Reserved.
```
This sentence doesn't apply nowadays

pedrobaeza commented on account_regularization/__init__.py:
```diff
@@ -0,0 +1,26 @@
+#    Copyright (c) 2008 Zikzakmedia S.L. (http://zikzakmedia.com)
+#  All Rights Reserved.
+#                       Jordi Esteve <jesteve@zikzakmedia.com>
+#    Copyright (c) 2008 Acysos SL. All Rights Reserved.
```
The same for the last part

pedrobaeza commented on account_regularization/__init__.py:
```diff
@@ -0,0 +1,26 @@
+#  All Rights Reserved.
+#                       Jordi Esteve <jesteve@zikzakmedia.com>
+#    Copyright (c) 2008 Acysos SL. All Rights Reserved.
+#    $Id$
```
Remove this line

pedrobaeza commented on account_regularization/__openerp__.py:
```diff
@@ -0,0 +1,55 @@
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (c) 2008 Zikzakmedia S.L. (http://zikzakmedia.com)
+#   All Rights Reserved.
```
The same

Roodin committed:
[FIX] account_regularization. Problem with view and delte account.py

pedrobaeza commented on account_regularization/__openerp__.py:
```diff
@@ -0,0 +1,55 @@
+{
+    "name": "Account Regularizations",
+    "version": "1.0",
+    "author": "ACYSOS S.L.",
```
Include you, ZikZak and "Odoo Community Association (OCA)" (for indexing in apps.odoo.com) as co-authors. See this as an example: https://github.com/odoomrp/odoomrp-wip/blob/8.0/quality_control/__openerp__.py#L27

pedrobaeza commented on account_regularization/__openerp__.py:
```diff
@@ -0,0 +1,55 @@
+    the 'Profit and Loss' regularization
+""",
+    "depends": ["account"],
+    "demo_xml": [],
```
This key is deprecated

pedrobaeza commented on account_regularization/i18n/es.po:
```diff
@@ -0,0 +1,201 @@
+#: field:wizard.regularize,create_uid:0
+msgid "Created by"
+msgstr "Creado por"
+s
```
This character screw the file

pedrobaeza commented on account_regularization/i18n/es.po:
```diff
@@ -0,0 +1,201 @@
+#. module: account_regularization
+#: field:wizard.regularize,fiscalyear:0
+msgid "Fiscal year"
+msgstr "Año fiscal"
```
s/Año/Ejercicio

pedrobaeza commented on account_regularization/models/__init__.py:
```diff
@@ -0,0 +1,27 @@
+#
+#    Odoo, Open Source Management Solution
+#    Copyright (c) 2008 Zikzakmedia S.L. (http://zikzakmedia.com)
+#    All Rights Reserved.
```
The same

pedrobaeza commented on account_regularization/models/account.py:
```diff
@@ -0,0 +1,168 @@
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (c) 2008 Zikzakmedia S.L. (http://zikzakmedia.com)
+#    All Rights Reserved.
```
The same

pedrobaeza commented on account_regularization/models/account.py:
```diff
@@ -0,0 +1,168 @@
+
+
+class AccountAccount(models.Model):
+    _name = 'account.account'
```
You don't need to repeat this variable

Roodin replied:
Still a lot of things to learn.
Thanks  for the review!!!

2015-06-26 19:56 GMT+02:00 Pedro M. Baeza notifications@github.com:

> In account_regularization/models/account.py
> https://github.com/OCA/account-financial-tools/pull/247#discussion_r33379171
> :
> 
> > +#    This program is distributed in the hope that it will be useful,
> > +#    but WITHOUT ANY WARRANTY; without even the implied warranty of
> > +#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
> > +#    GNU General Public License for more details.
> > +#
> > +#    You should have received a copy of the GNU General Public License
> > +#    along with this program.  If not, see http://www.gnu.org/licenses/.
> > +#
> > +##############################################################################
> > +
> > +from openerp import models, api, fields
> > +import time
> > +
> > +
> > +class AccountAccount(models.Model):
> > -    _name = 'account.account'
> 
> You don't need to repeat this variable
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/OCA/account-financial-tools/pull/247/files#r33379171.

pedrobaeza commented on account_regularization/models/account.py:
```diff
@@ -0,0 +1,168 @@
+        query = self.env['account.move.line']._query_get()
+
+        if not periods:
+            self.env.cr.execute(("SELECT a.id, "
```
Don't use SQL

pedrobaeza commented on account_regularization/models/account.py:
```diff
@@ -0,0 +1,168 @@
+        else:
+            period_ids = ",".join(str(period.id) for period in periods)
+            self.env.cr.execute(("SELECT a.id, "
+                                 "COALESCE(SUM(l.debit - l.credit), 0) "
```
Don't use SQL

Roodin replied:
Yes I Know, but no time at this moment for this. So I promise to do this in a next PR

pedrobaeza commented on account_regularization/models/account.py:
```diff
@@ -0,0 +1,168 @@
+    _description = 'Account Regularization Model'
+
+    name = fields.Char('Name', size=32, required=True)
+    account_ids = fields.Many2many('account.account',
```
You can use this layout to be less constrained:

```
account_ids = fields.Many2many(
    'account.account', 'account_regularization_rel', 'regularization_id',
    'account_id', 'Accounts to balance', required=True, 
    domain=[('type', '=', 'view')])
```

pedrobaeza commented on account_regularization/models/account.py:
```diff
@@ -0,0 +1,168 @@
+            'period_id': period.id,
+        }
+        self.env['account.move.line'].create(diff_line)
+        return
```
You don't need this sentence

pedrobaeza commented on account_regularization/models/account.py:
```diff
@@ -0,0 +1,168 @@
+
+
+class AccountMove(models.Model):
+    _name = 'account.move'
```
Don't put this

pedrobaeza commented on account_regularization/models/account.py:
```diff
@@ -0,0 +1,168 @@
+                browse(tot_account_ids).balance_calculation(periods=period_ids)
+        if balance_results.keys().__len__() ==\
+                balance_results.values().count(0.0):
+            raise Exception('Nothing to regularize',
```
use openerp.exceptions.Warning

pedrobaeza commented on account_regularization/models/account.py:
```diff
@@ -0,0 +1,168 @@
+        #regularization = self.browse(cr, uid, id)[0]
+        # Find all children accounts
+        account_ids = [x.id for x in self.account_ids]
+        tot_account_ids = self.env['account.account'].\
```
You don't need to query childrens, as you can make a move with an account view. Just a call to browse with the corresponding context keys will return in balance/credit/debit fields the proper value

Roodin replied:
@pedrobaeza I don't understand this one. I need to query childrens because I need the balance on each child account  and make a move on each of them with balance != 0

pedrobaeza commented on account_regularization/models/account_regularization.py:
```diff
@@ -0,0 +1,168 @@
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (c) 2008 Zikzakmedia S.L. (http://zikzakmedia.com)
+#    All Rights Reserved.
```
The same

pedrobaeza commented on account_regularization/models/account_regularization.py:
```diff
@@ -0,0 +1,168 @@
+
+
+class AccountAccount(models.Model):
+    _name = 'account.account'
```
Remove this

pedrobaeza commented on account_regularization/models/wizard_regularize.py:
```diff
@@ -0,0 +1,90 @@
+# -*- encoding: utf-8 -*-
```
Put it on wizard folder

pedrobaeza commented on account_regularization/models/wizard_regularize.py:
```diff
@@ -0,0 +1,90 @@
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (c) 2008 Zikzakmedia S.L. (http://zikzakmedia.com)
+#    All Rights Reserved.
```
The same

pedrobaeza commented on account_regularization/static/description/icon.svg:
```diff
@@ -0,0 +1,79 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
As you haven't modified the icon, it's not needed to include the SVG

pedrobaeza commented on account_regularization/views/account_regularization_view.xml:
```diff
@@ -0,0 +1,43 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+	<data>
```
You have to use spaces instead of tabs

pedrobaeza commented on account_regularization/views/wizard_regularization_view.xml:
```diff
@@ -0,0 +1,37 @@
+<openerp>
+    <data>
+        <record id='view_wizard_regularization_form' model='ir.ui.view'>
+			<field name='name'>wizard.regularization.form</field>
```
Use spaces

Roodin committed:
[FIX] account_regularization. Several fixes from PR comments

Roodin committed:
[FIX] account_regularization. Moved wizard to directory

Roodin committed:
[FIX] account_regularization. Added README.rst

Roodin committed:
[FIX]  account_regularization.  Fixed error with created move date and other improvements in code

pedrobaeza commented:
@Roodin are you still interested in this?

Roodin commented:
Not sure.  Probably no. A long time ago...

pedrobaeza commented:
OK, closing. Tell to reopen it if you come back to this.
